### PR TITLE
Add CopyConstness

### PR DESCRIPTION
**Pull request by MetaLang**
This builds on @Hackerpilot's previous PR for [CopyStorageClasses](https://github.com/D-Programming-Language/phobos/pull/2782). It adds a template that will only copy the "constness" of a type, leaving it un-`shared` even if the source type is `shared`. This is a companion PR to https://github.com/D-Programming-Language/phobos/pull/3442.

### CODE REVIEW

File: std/traits.d [hunk 1]
```diff
@@ -6018,6 +6019,79 @@ unittest
...
+    //But the constness will be
+    assert(is(CopyConstness!(shared const real, double) == const double));
+
+    //Careful, const(int)[] is a mutable array of const(int)
+    alias MutT = CopyConstness!(const(int)[], int);
+    assert(!is(MutT == const(int)));
+
+    //Okay, const(int[]) applies to array and contained ints
+    alias CstT = CopyConstness!(const(int[]), int);
+    assert(is(CstT == const(int)));
```
Comment (burner) [inline std/traits.d]: could you add a space between "(is" if it is the opposite to "(!is"

that makes it easier to read IMO.
lilke
´´´d
alias MutT = CopyConstness!(const(int)[], int);
assert(!is(MutT == const(int)));

//Okay, const(int[]) applies to array and contained ints
alias CstT = CopyConstness!(const(int[]), int);
assert( is(CstT == const(int)));
```

**Comment by burner** [comment]
other then my one comment LGTM

File: std/traits.d (hunk 1 above)
Comment (MetaLang) [inline std/traits.d]: Will do.

**Comment by DmitryOlshansky** [comment]
Looks fine

**Comment by burner** [comment]
there is some bad white space

:%s/\t/    /g
:%s/[ ]\+$//g

**Comment by MetaLang** [comment]
@burner I'm not seeing this rogue whitespace.

**Comment by burner** [comment]
@MetaLang it was at

Error - file 'std/traits.d' contains trailing whitespace at line 6023

**Comment by burner** [comment]
Auto-merge toggled on

**Comment by MetaLang** [comment]
@burner Ah, okay, I got rid of that when I changed the spacing in the assert statements.

**Pull request merged**